src/asymmetric_crypto/curve_25519/ed25519/public_key.rs

```diff
@@ -1,16 +1,19 @@
 //reexport the RustBox Ed25519 impl
 
+use std::ops::Deref;
+
+pub use ed25519_dalek::{PublicKey as EdPublicKey, SecretKey as EdSecretKey};
+
 use super::private_key::Ed25519PrivateKey;
 use crate::{CBytes, FixedSizeCBytes};
-pub use ed25519_dalek::{PublicKey as EdPublicKey, SecretKey as EdSecretKey};
-use std::ops::Deref;
 
 #[derive(Clone, PartialEq, Eq, Debug)]
 pub struct Ed25519PublicKey(pub(crate) EdPublicKey);
 
 impl CBytes for Ed25519PublicKey {}
 
 impl Ed25519PublicKey {
+    #[must_use]
```

Comment:
You don't need the `must_use` since this is a cheap operation that does not consumes `self`.
```suggestion
```